### DISCUSSION TITLE
Download image to Storage from remote url triggered by Postgres Triggers

### DISCUSSION THREAD

**Discussion by wadclapp** [supabase/supabase#12685] [Questions]
I'm using an API that returns a PNG (e.g. https://example.com/api?api_key=xxxx&opt1=x&opt2=y).

Is it possible using the Storage HTTP API, to ask it to download the file from the remote url into a folder?

I got a postgres plpgsql rpc function working that gets the image (using pgsql-http) and have it as a `bytea` in hexadecimal form - but I don't think it's possible to then get that in the right format to POST it in the body to storage and have it saved as a PNG? I can get it in base64 using `encode(textsend(content), 'base64')` but the storage API doesn't accept base64. I also don't know if trying to do it this way is a bad idea??

**Comment by wadclapp** [comment]
Doesn't seem possible with the storage api. Although would be great if it was, as now I will have to download it elsewhere on a different server just to re-upload it from there to supabase storage.

**Reply by ioucyf** [reply]
I'm working on trying to figure out exactly that!! That would be amazing if it'd be possible to do so with PostgreSQL functions!
